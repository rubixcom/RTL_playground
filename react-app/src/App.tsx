import React from "react";
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
import Calculator from "./Calculator"
import Chat from "./Chat"

export default function App() {

  return (
    <Router>
      <div>
        <nav>
            <Link to="/calculator">Calculator</Link>
                  <span style={{ padding: '50px' }}>&nbsp;</span>
            <Link to="/chat">Chat</Link>
        </nav>

        {/* A <Switch> looks through its children <Route>s and
            renders the first one that matches the current URL. */}
        <Switch>
          <Route path="/calculator">
            <Calculator />
            </Route>      
          <Route path="/chat">
            <Chat />
          </Route>
        </Switch>
      </div>
    </Router>
  );
}
