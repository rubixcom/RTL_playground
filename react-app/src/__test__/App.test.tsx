import React from 'react';
import { render, cleanup, fireEvent, getByLabelText } from "@testing-library/react";
import "@testing-library/jest-dom";
import App from '../App';

afterEach(cleanup);

it("matches snapshot", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});