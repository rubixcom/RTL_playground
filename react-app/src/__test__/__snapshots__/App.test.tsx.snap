// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav>
      <a
        href="/calculator"
      >
        Calculator
      </a>
      <span
        style="padding: 50px;"
      >
        Â 
      </span>
      <a
        href="/chat"
      >
        Chat
      </a>
    </nav>
  </div>
</DocumentFragment>
`;
